openssl/src/ssl/mod.rs

```diff
@@ -307,6 +307,39 @@ extern "C" fn raw_verify<F>(preverify_ok: c_int, x509_ctx: *mut ffi::X509_STORE_
     }
 }
 
+extern "C" fn raw_psk<F>(ssl: *mut ffi::SSL,
+                         hint: *const c_char,
+                         identity: *mut c_char,
+                         _max_identity_len: c_uint,
+                         psk: *mut c_uchar,
+                         _max_psk_len: c_uint) -> c_uint
+    where F: Fn(&mut SslRef, &str) -> Result<(String, Vec<u8>), ErrorStack> + Any + 'static + Sync + Send
```

Comment:
I would probably expose the OpenSSL API more directly - something like `Fn(&mut SslRef, hint: Option<&[u8]>, identity: &mut [u8], psk: &mut [u8]) -> Result<(usize, usize), ErrorStack>`.